### PR TITLE
Activate Travis tests and add authentication parameters

### PR DESCRIPTION
**Pull request by blag**
This PR activates Travis tests for this repository. Travis is free (for this project) and easy to use, so it's easier to configure, extend, and fix when it breaks.

I added alternative authentication parameters for multiple mixins. This allows users to specify an object that contains a `token` or `key` key/value that overrides the client's stored authentication token/API key.

I also add tests for how to use auth tokens and API keys, add unrelated tests for the `Schedulable` mixin, and additional tests all around to test the alternative authentication parameters for all relevant mixins.

I updated the `nock` dependency to fix an issue where the nock mock was not properly matching requests based on a header value.

Coverage went from:

```
-------------------------|----------|----------|----------|----------|-------------------|
File                     |  % Stmts | % Branch |  % Funcs |  % Lines | Uncovered Line #s |
-------------------------|----------|----------|----------|----------|-------------------|
All files                |    86.16 |    81.62 |    81.67 |    86.16 |                   |
 st2client.js/lib        |    86.36 |       75 |      100 |    86.36 |                   |
  endpoint.js            |       95 |     87.5 |      100 |       95 |                23 |
 st2client.js/lib/mixins |    83.96 |    83.13 |     81.4 |    83.96 |                   |
  attributable.js        |      100 |     87.5 |      100 |      100 |                29 |
  authenticatable.js     |    93.75 |     87.5 |      100 |    93.75 |                 7 |
  deletable.js           |      100 |     87.5 |      100 |      100 |                22 |
  editable.js            |      100 |      100 |      100 |      100 |                   |
  enumerable.js          |      100 |      100 |      100 |      100 |                   |
  paginatable.js         |      100 |    94.12 |      100 |      100 |                 3 |
  readable.js            |      100 |    83.33 |      100 |      100 |                23 |
  repeatable.js          |      100 |      100 |      100 |      100 |                   |
  routable.js            |       80 |    83.33 |      100 |       80 |              7,16 |
  schedulable.js         |    33.33 |        0 |        0 |    33.33 |        6,11,12,19 |
  streamable.js          |       50 |    62.86 |       40 |       50 |... ,97,99,102,104 |
  watchable.js           |    96.43 |    93.75 |      100 |    96.43 |                42 |
  writable.js            |    90.91 |     87.5 |      100 |    90.91 |                25 |
-------------------------|----------|----------|----------|----------|-------------------|
```

to this:

```
-------------------------|----------|----------|----------|----------|-------------------|
File                     |  % Stmts | % Branch |  % Funcs |  % Lines | Uncovered Line #s |
-------------------------|----------|----------|----------|----------|-------------------|
All files                |    88.37 |    83.33 |    85.25 |    88.37 |                   |
 st2client.js/lib        |    85.48 |    70.83 |      100 |    85.48 |                   |
  endpoint.js            |    93.75 |       75 |      100 |    93.75 |                19 |
 st2client.js/lib/mixins |    88.02 |    86.63 |    86.36 |    88.02 |                   |
  attributable.js        |      100 |     87.5 |      100 |      100 |                33 |
  authenticatable.js     |    93.75 |     87.5 |      100 |    93.75 |                 7 |
  deletable.js           |      100 |     87.5 |      100 |      100 |                26 |
  editable.js            |      100 |      100 |      100 |      100 |                   |
  enumerable.js          |      100 |      100 |      100 |      100 |                   |
  paginatable.js         |      100 |    94.12 |      100 |      100 |                 3 |
  readable.js            |      100 |    83.33 |      100 |      100 |                27 |
  repeatable.js          |      100 |      100 |      100 |      100 |                   |
  routable.js            |       80 |    83.33 |      100 |       80 |              7,16 |
  schedulable.js         |      100 |      100 |      100 |      100 |                   |
  streamable.js          |       50 |    62.86 |       40 |       50 |... ,97,99,102,104 |
  util.js                |      100 |      100 |      100 |      100 |                   |
  watchable.js           |    96.43 |    93.75 |      100 |    96.43 |                42 |
  writable.js            |    92.31 |     87.5 |      100 |    92.31 |                29 |
-------------------------|----------|----------|----------|----------|-------------------|
```

(irrelevant coverage results have been truncated)

### CODE REVIEW

File: lib/mixins/attributable.js [hunk 1]
```diff
@@ -17,10 +17,22 @@ var Attributable = {
...
+
+      if (alt_auth && alt_auth.token) {
+        request_opts.headers = {
+          'x-auth-token': alt_auth.token
+        };
+      } else if (alt_auth && alt_auth.api_key) {
+        request_opts.headers = {
+          'st2-api-key': alt_auth.api_key
+        };
+      }
```
Comment (m4dcoder) [inline lib/mixins/attributable.js]: There's a lot of duplication of this code block sprinkle across the code base here.  Is it possible to put this in a reusable function?

File: lib/mixins/enumerable.js [hunk 2]
```diff
@@ -2,16 +2,28 @@
...
+      if (alt_auth && alt_auth.token) {
+        request_opts.headers = {
+          'x-auth-token': alt_auth.token
+        };
+      } else if (alt_auth && alt_auth.api_key) {
+        request_opts.headers = {
+          'st2-api-key': alt_auth.api_key
+        };
+      }
+
```
Comment (jinpingh) [inline lib/mixins/enumerable.js]: @blag If this part of code is duplicated in almost all mixins, could it be a function call in util or other file?

File: lib/mixins/attributable.js (hunk 1 above)
Comment (blag) [reply lib/mixins/attributable.js]: Fixed.

File: lib/mixins/enumerable.js (hunk 2 above)
Comment (blag) [reply lib/mixins/enumerable.js]: Fixed.

**Pull request merged**